### ISSUE TITLE
vue/require-valid-default-prop demands 'string', even when prop type is 'symbol | string'

### ISSUE THREAD

**Issue by femans** [vuejs/eslint-plugin-vue#2174]
**Checklist**
- [x ] I have tried restarting my IDE and the issue persists.
- [x ] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** eslint@8.34.0
- **eslint-plugin-vue version:** eslint-plugin-vue@9.9.0
- **Node version:** v18.13.0
- **Operating System:** Ubuntu 22.04LTS

**Please show your full configuration:**

```json5
/* eslint-env node */
require("@rushstack/eslint-patch/modern-module-resolution");

module.exports = {
  root: true,
  extends: [
    "plugin:vue/vue3-essential",
    "eslint:recommended",
    "@vue/eslint-config-typescript",
    "@vue/eslint-config-prettier",
  ],
  parserOptions: {
    ecmaVersion: "latest",
  },
};

```

**What did you do?**	
```vue
export type ArrangeableProps<PayloadType extends object> = {
  name?: symbol | string;
// ...
};

const props = withDefaults(defineProps<ArrangeableProps<object>>(), {
  name: Symbol(),
  // ...
});
```

**What did you expect to happen?**
I need the default to be a symbol, because it has to be a unique identifier. But the consumer-code should be able to give a unique string as a name. The code runs fine as I wrote it, and I expect no linting errors.

**What actually happened?**
I get the linting error from the title. `Type of the default value for 'name' prop must be a string.eslint[vue/require-valid-default-prop](https://eslint.vuejs.org/rules/require-valid-default-prop.html)`
N.b. it seems to be typescript related, because a quick test doing this with JS does not produce the same error.

**Repository to reproduce this issue**
https://github.com/vuereka/vue-arrange/

**Comment by ota-meshi** [comment]
Thank you for the report!

I have confirmed the bug you reported.
However, it seems to have been (unintentionally) fixed in the latest version we just released.
https://github.com/vuejs/eslint-plugin-vue/releases/tag/v9.13.0

Please upgrade the plugin to the latest version.

**Comment by femans** [comment]
Seems you are right! Thanks!
